pkg/cloudinit/common.go

```diff
@@ -78,14 +80,24 @@ func NewBaseCloudConfig(data BaseUserData) (CloudConfig, error) {
 			Owner:       "root:root",
 		})
 	}
+
+	var configFileContents string
+	switch {
+	case data.BootstrapConfig != "":
```

Comment:
why switch case and not a simple if block?

Reply:
Go linter barked at me once to change an if to a switch and i have been traumatized ever since

Reply:
Ok, I played around with golangci yesterday and it was barking at me the other way around so you might be barked at again if we introduce that here.
But feel free to ignore it until then - I don't have a strong opinion (here and in general).